modules/framework/galasa-parent/dev.galasa.framework.api.users/src/main/java/dev/galasa/framework/api/users/internal/routes/BeanTransformer.java

```diff
@@ -62,9 +70,54 @@ public UserData convertUserToUserBean(IUser userIn) {
         clientsOut = clientsOutList.toArray(clientsOut);
         userOut.setclients(clientsOut);
 
+        String roleId = getRoleId(userIn);
+        userOut.setrole(roleId);
+
+        UserSynthetics synthetic = createSynthetic(roleId, this.baseUrl);
+        userOut.setsynthetic(synthetic);
+
         return userOut ;
     }
 
+    private String getRoleId( IUser userIn ) throws InternalServletException {
+        String roleId = userIn.getRoleId();
+        if (roleId==null || roleId.trim().equals("")) {
+            try {
+                roleId = rbacService.getDefaultRoleId();
+            } catch( RBACException ex) {
+                ServletError err = new ServletError(GAL5086_FAILED_TO_GET_DEFAULT_ROLE);
+                throw new InternalServletException(err, 500, ex);
```

Comment:
Really minor - Could use `HttpServletResponse.SC_INTERNAL_SERVER_ERROR` instead of `500` here

Reply:
fixed.